src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java

```diff
@@ -252,6 +248,10 @@ public Config validate(Map<String, String> connectorConfigs)
             ": proxy username and password must be provided together");
       }
     }
+    // If private key or private key passphrase is provided through file, skip validation
+    if (connectorConfigs.getOrDefault(Utils.SF_PRIVATE_KEY, "").contains("${file:") ||
+      connectorConfigs.getOrDefault(Utils.PRIVATE_KEY_PASSPHRASE, "").contains("${file:"))
+      return result;
```

Comment:
I am ok on skipping validation in this case. please make sure, reading from file works.

Reply:
Reading from file is handled by the Kafka Connect framework. I think they should send us the "parsed" configuration in this validation api, but they are not. File references in the config are not properly read & replaces with content in the file. 
But when they start the connector, the config is replaced with the content of the file.